### PR TITLE
Loading state for buttons #430

### PR DESCRIPTION
**Pull request by imjaglan**
### changes :  Enhancement  #430
- Added the loading state prop
- Button disabled on loading and it's styles

### CODE REVIEW

**Comment by niwsa** [comment]
Thank you @imjaglan for the PR. I will review this shortly.

File: src/shared/Button/index.lite.tsx [hunk 1]
```diff
@@ -14,7 +14,24 @@ export default function Button(props: ButtonProps) {
...
+          width='24'
+          height='24'
+          className={`${styles.loading}`}
+          viewBox='0 0 24 24'
+          fill='none'
+          stroke='currentColor'
+          stroke-width='2'
+          stroke-linecap='round'
+          stroke-linejoin='round'
+          class='lucide lucide-loader-2'>
```
Comment (niwsa) [inline src/shared/Button/index.lite.tsx:31]: Do we need these classes?

Comment (imjaglan) [reply src/shared/Button/index.lite.tsx:31]: It's the loading svg, we can use from a icon library or you want me to create a custom one?

Comment (niwsa) [reply src/shared/Button/index.lite.tsx:31]: Ahh, I could have been more specific here. So we already have a spinner here: https://github.com/boxyhq/ui/blob/148aff4a8de84a22c9e9d3095ae110126c03e2b3/src/shared/LoadingContainer/index.lite.tsx#L8. What you can do is create a new `Spinner` component (inside shared folder) by copying just this spinner div and its linked CSS. That way we can use it in both places.

**Comment by imjaglan** [comment]
Hey @niwsa  is something wrong with the code? I'm not able to figure it out.

**Comment by Ryukemeister** [comment]
> Hey @niwsa is something wrong with the code? I'm not able to figure it out.

Hey @imjaglan, can you try running the command [npm run build](https://github.com/boxyhq/ui/blob/main/package.json#L14) from the rootof your project? That will generate all the required mitosis files, you can find them under the `src` folder for every framwork folder, eg: for react the generated files for react would be generated under [react/src](https://github.com/boxyhq/ui/tree/main/react/src) and then you can check the generated react code for the files that you've created in order to see if there's something wrong with it.

**Comment by niwsa** [comment]
@imjaglan The mitosis build looks good now... The failures you see in Angular/Vue builds are expected. As @Ryukemeister  mentioned you can also debug issues locally. I didn't get around to reviewing the new changes but will be on it soon.

**Comment by niwsa** [comment]
@imjaglan Overall LGTM. One thing we can do is default to `aria-hidden='true'` for Spinner and wrap it under a div inside `LoadingContainer`. I will make the changes and pull this in.

**Review by niwsa — APPROVED** [review]

**Pull request merged**